Testscripts/Linux/validate-da.sh

```diff
@@ -0,0 +1,243 @@
+#!/usr/bin/env bash
+#######################################################################
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the Apache License.
+#
+# validate-da.sh
+# Description:
+#    Validate the installation and uninstallation of the Dependency Agent.
+#    Validate the enable and disable of the Dependency Agent.
+#######################################################################
+
+set -e
+set -x
+
+# Source utils.sh
+. utils.sh || {
+    echo "ERROR: unable to source utils.sh!"
+    echo "TestAborted" > state.txt
+    exit 0
+}
+
+# Get distro information
+GetDistro
+
+dir_list=(/opt/microsoft/dependency-agent /var/opt/microsoft/dependency-agent /etc/opt/microsoft/dependency-agent /lib/microsoft-dependency-agent /etc/init.d/microsoft-dependency-agent)
+supported_distro_list=(redhat centos suse debian ubuntu)
+
+readonly da_pid_file="/etc/opt/microsoft/dependency-agent/config/DA_PID"
+
+function test_case_cleanup(){
+    /opt/microsoft/dependency-agent/uninstall
+
+    for dir in "${dir_list[@]}"
+    do
+        rm -rf $dir
+    done
+
+    kill $watch_pid
```

Comment:
Please add a comment that if the watch process hasn't been started, then `$watch_pid` will be unset and the kill will be a NOP. It would be very easy for someone maintaining the code in the future to think "Hey! That's not good coding, using a variable before it is set! Let me put in a line of code to initialize it to ..."

OR

Do initialize `watch_pid` to `""` with a comment that there are no processes to kill initially.

Reply:
Added the comment